pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagentplan.go

```diff
@@ -56,3 +83,81 @@ func (h *handler) OnChangeInstallSUC(cluster *rancherv1.Cluster, status rancherv
 		mcc,
 	}, status, nil
 }
+
+// syncSystemUpgradeControllerStatus queries the managed system-upgrade-controller chart and determines if it is properly configured for a given
+// version of Kubernetes. It applies a condition onto the control-plane object to be used by the planner when handling Kubernetes upgrades.
+func (h *handler) syncSystemUpgradeControllerStatus(obj *rkev1.RKEControlPlane, status rkev1.RKEControlPlaneStatus) (rkev1.RKEControlPlaneStatus, error) {
+	// perform the same name limiting as in the OnChangeInstallSUC controller, but prepend the 'mcc-' prefix that is added when the bundle is created
+	bundleName := fmt.Sprintf("mcc-%s", name.Limit(name.SafeConcatName(obj.Name, "managed", "system-upgrade-controller"), 48))
+	sucBundle, err := h.bundles.Get(fleet.ClustersDefaultNamespace, bundleName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		// if we couldn't find the bundle then we know it's not ready
+		rke2.SystemUpgradeControllerReady.False(&status)
+		// don't return the error, otherwise the status won't be set to 'false'
+		return status, nil
+	}
+
+	if err != nil {
+		return status, err
+	}
+
+	rke2.SystemUpgradeControllerReady.Unknown(&status)
+
+	// determine if the SUC deployment has been rolled out fully, and if there were any errors encountered
+	if sucBundle.Status.Summary.Ready != sucBundle.Status.Summary.DesiredReady {
+		if sucBundle.Status.Summary.ErrApplied != 0 && len(sucBundle.Status.Summary.NonReadyResources) > 0 {
+			nonReady := sucBundle.Status.Summary.NonReadyResources
+			rke2.SystemUpgradeControllerReady.Reason(&status, fmt.Sprintf("Error Encountered Waiting for System Upgrade Controller Deployment To Roll Out: %s", nonReady[0].Message))
```

Comment:
nit: capitalization here is a little wonky, it can be written as a sentence. Not important though